semgrep-core/src/engine/Match_extract_mode.ml

```diff
@@ -293,12 +302,18 @@ let extract_and_concat erule_table xtarget rule_ids matches =
          |> List.rev
          (* Read the extracted text from the source file *)
          |> Common.map (fun { start_pos; start_line; start_col; end_pos } ->
-                let contents =
+                let contents_raw =
                   Common.with_open_infile xtarget.Xtarget.file (fun chan ->
                       let extract_size = end_pos - start_pos in
                       seek_in chan start_pos;
                       really_input_string chan extract_size)
                 in
+                (* Convert from JSON to plaintext, if required *)
```

Comment:
add a TODO like:
TODO: note that this extract_code_... mess up the position information so match on the
extract code would not correspond to precise location in the original file.

Reply:
In theory that is managed by the previous extract mode feature. I just added a fix in that location for the JSON decoding.

Reply:
At a glance, I don't think that adjustment is robust enough to accurately handle what it would need to, but I'm not in a good position to test. Ideally it shouldn't rely on the code array just happening to be formatted in such a way each string is on it's own line, etc..

Are columns also not slightly off in the instances where we needed to delete characters as part of the unescaping?

Reply:
Yeah, columns are not very accurate. @aryx mentioned that shouldn't be a problem right now.